src/components/MapsSection/MapsSection.tsx

```diff
@@ -102,6 +104,57 @@ const MapsSection = ({
     });
   }, [currentVisu]);
 
+  useEffect(() => {
+    const button = buttonRef.current;
+    const container = tagsContainerRef.current;
+
+    if (container && button) {
+      const currentIndex = tiffInfo.findIndex(
+        (map) => map.fields.id === currentVisu.id,
+      );
+
+      const containerTop = container.scrollTop;
+      const containerBottom = containerTop + container.clientHeight;
+
+      const buttonOffsetHeight = button.offsetHeight;
+      const buttonHeight = button.offsetTop;
+      +buttonOffsetHeight;
+
+      let tamBotao = buttonHeight / 100;
+      const isMobile = window.innerWidth <= 810;
+      const screen = window.innerHeight;
+      const maxVisibleBoxes = isMobile ? 1 : 5;
+      const needsScroll =
+        (buttonHeight / 9) * currentIndex > containerBottom ||
```

Comment:
How did you came up with this solution? Just curious

Reply:
Analyzing the values returned by the functions and looking for a pattern that would show me when a button needs scrolling, i mean, when it overflows the container at the moment it is viewed. Basically, **buttonHeight * currentIndex** represents the size of the button multiplied by the number of buttons **behind it** (if it's the first button, currentIndex = 0; if it's the second button, currentIndex = 1 because there is one button behind it). So, if this size is greater than the container's size, it means it has overflowed, in short, **it needs scrolling when the sum of the button sizes exceeds the containerBottom**. Since buttonHeight didn’t return an exactly suitable value, I had to adjust it by dividing by 9, this part involved a bit of trial and error. I  also tried using some other button height functions, but they weren't as effective.